frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx

```diff
@@ -0,0 +1,102 @@
+import './BulkInviteModal.scss'
+import { capitalizeFirstLetter, isEmail } from 'lib/utils'
+import { bulkInviteLogic } from './bulkInviteLogic'
+
+const PLACEHOLDER_NAMES = ['jane', 'john']
+
+function InviteRow({ index }: { index: number }): JSX.Element {
+    const name = PLACEHOLDER_NAMES[index % 2]
+    const { invites } = useValues(bulkInviteLogic)
+    const { updateInviteAtIndex } = useActions(bulkInviteLogic)
+
+    return (
+        <Row gutter={16} className="invite-row">
+            <Col xs={12}>
+                <Input
+                    placeholder={`${name}@posthog.com`}
+                    type="email"
+                    className={`error-on-blur${!invites[index].isValid ? ' errored' : ''}`}
+                    onChange={(e) => {
+                        const { value } = e.target
+                        let isValid = true
+                        if (value && !isEmail(value)) {
+                            isValid = false
+                        }
+                        updateInviteAtIndex({ email: e.target.value, isValid }, index)
+                    }}
+                    value={invites[index].email}
+                />
+            </Col>
+            <Col xs={12}>
+                <Input
+                    placeholder={capitalizeFirstLetter(name)}
+                    onBlur={(e) => {
+                        updateInviteAtIndex({ first_name: e.target.value }, index)
+                    }}
+                />
+            </Col>
+        </Row>
+    )
+}
+
+export function BulkInviteModal({ visible, onClose }: { visible: boolean; onClose: () => void }): JSX.Element {
+    const { user } = useValues(userLogic)
+    const { invites, canSubmit, invitedTeamMembersLoading, invitedTeamMembers } = useValues(bulkInviteLogic)
+    const { addMoreInvites, resetInvites, inviteTeamMembers } = useActions(bulkInviteLogic)
+
+    useEffect(() => {
+        if (invitedTeamMembers.invites.length) {
+            onClose()
+        }
+    }, [invitedTeamMembers])
+
+    return (
+        <>
+            <Modal
+                title={`Invite your team members${user?.organization ? ' to ' + user?.organization?.name : ''}`}
+                visible={visible}
+                onCancel={() => {
+                    resetInvites()
+                    onClose()
+                }}
+                onOk={inviteTeamMembers}
+                okText="Invite team members"
+                destroyOnClose
+                okButtonProps={{ disabled: !canSubmit, loading: invitedTeamMembersLoading }}
+                cancelButtonProps={{ disabled: invitedTeamMembersLoading }}
+                closable={!invitedTeamMembersLoading}
+            >
+                <div className="bulk-invite-modal">
+                    <div>
+                        Invite as many team members as you want. <b>Names are optional</b>, but it will speed up the
+                        process for your teammates.
+                    </div>
```

Comment:
So as far as I understand, this is to make the experience smoother for the invite. Can the invitee deviate from this optionally preprovided name?

Reply:
Yes, that's the idea! Not yet, but #2734 introduces support for that